src/Stripe.Tests.XUnit/Services/Coupons/StripeCouponCreateOptionsTest.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+using Stripe.Infrastructure;
+
+namespace Stripe.Tests.Xunit
+{
+    public class StripeCouponCreateOptionsTest : BaseStripeTest
+    {
+        public StripeCouponService Service;
+
+        public StripeCouponCreateOptionsTest() {
+            this.Service = new StripeCouponService();
+        }
+
+        [Fact]
+        public void Serialize()
+        {
+            var options = new StripeCouponCreateOptions() {
+                PercentOff = 25,
+                Duration = "forever",
+            };
+
+            var url = this.Service.ApplyAllParameters(options, "", false);
+            Assert.Equal("?percent_off=25&duration=forever", url);
```

Comment:
I like this approach but not entirely sure we want this for every single resource. I think something like what we have in Go to test all "types" could work better.
Otherwise we likely want to only test complex types like lists? We also should test all options class otherwise so create and update here.

Reply:
Right, to be clear, this particular test is fairly useless and I completely agree that most of this stuff should be tested only once on `ParameterBuilder`. This was just an example of what a custom serialization test could work. In practice, we'd probably only want these for classes that use custom plugins.